projects/solar/test/test_fan_control_solar.c

```diff
@@ -0,0 +1,234 @@
+#include "fan_control_solar.h"
+
+#include <stdbool.h>
+
+#include "data_store.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "fault_handler.h"
+#include "fault_monitor.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "max6643_fan_controller.h"
+#include "ms_test_helpers.h"
+#include "solar_boards.h"
+#include "solar_events.h"
+#include "spv1020_mppt.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+// Any defines here
+
+// Declare static test variables
+static GpioAddress s_overtemp_addr;
+static GpioAddress s_fan_fail_addr;
+static GpioAddress s_full_speed_addr;
+static uint16_t s_full_speed_temp_threshold;
+static SolarMpptCount s_mppt_count;
+
+static GpioState s_gpio_state;
+static Event s_e;
+
+static GpioAddress s_dummy;
+
+static void TEST_MOCK(prv_fanfail_callback)(const GpioAddress *address, void *context) {
+  LOG_WARN("fan_control detected fanfail, raising fault event");
+  fault_handler_raise_fault(EE_SOLAR_FAULT_FAN_FAIL, 0);
+}
+
+void setup_test(void) {
+  event_queue_init();
+  gpio_init();
+  gpio_it_init();
+  data_store_init();
+
+  // Test Board A with 6 MPPTs uses virtual port A
+  GpioAddress s_overtemp_addr = { .port = GPIO_PORT_A, .pin = 0 };
+  GpioAddress s_fan_fail_addr = { .port = GPIO_PORT_A, .pin = 1 };
+  GpioAddress s_full_speed_addr = { .port = GPIO_PORT_A, .pin = 2 };
+
+  // arbitrarily selected value for testing
+  uint16_t s_full_speed_temp_threshold = 3;
+
+  SolarMpptCount s_mppt_count = SOLAR_BOARD_6_MPPTS;
```

Comment:
This isn't setting the values of the static variables, it's declaring local variables of the same name that shadow the global variables - don't use the type name if you want to reset the variables. But I think these are meant to be constants, so you can just set them when you declare them at the top level, and then you don't have to worry about setting them in setup_test.